### DISCUSSION TITLE
Welcome to hirob-be-core Discussions!

### DISCUSSION THREAD

**Discussion by husamettinarabaci** [Ubn-Jr/hirob-be-core#10] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Sddilora** [comment]
Hello everyone! 👋

I'm Dilara, the leader of the back-end team, and our responsibility is the back-end aspect of this project. We play a crucial role in the project's overall functionality, so it's essential to maintain continuous communication with all the groups until the project's completion.

Now, let's explore the key topics we should discuss, and please feel free to share any additional points that come to mind:

**Expectations and Requirements:** We need to clarify what each team expects from us in terms of functionality, performance, and deliverables. This includes understanding the specific needs of the @Ubn-Jr/hirob-em , as well as the @Ubn-Jr/hirob-fe and @Ubn-Jr/hirob-mo 

**Integration Points:** Discuss how our back-end systems will integrate with the front-end and other services. Are there any specific APIs or data formats that need to be adhered to for seamless integration?

**Data Models and Database:** Determine the data models required for our project and plan the database structure accordingly. Discuss data storage, retrieval, and data processing needs.

**Documentation:** Emphasize the importance of documentation for our back-end code, APIs, and any technical processes or guidelines we need to follow.

**Comment by Sddilora** [comment]
Let me start; here are the ideas that have come to my mind.

We need to develop endpoints and handlers for the following functionalities:
- HiROB's actions (1 left, 1 straight, 1 back, 1 left)
- HiROB's map (playground) for both static maps and maps created by users.

Additionally, we require a robust map algorithm. Our maps will consist of individual pieces, and it's crucial to devise an algorithm that allows users to place each piece freely, while our system adapts seamlessly to these changes.

**Comment by Sddilora** [comment]
Please follow the repository's issues and discussions and contribute your insights, dear @gnsamine, @mervegonc, @furkanadiiguzel, @johsquaree, and @Feyzanrs.
